### PR TITLE
Add delete route to favorites controller

### LINKED ISSUE #14: Epic - Favorite Locations
## Description
As a registered user, when I want to be able to do the following with my favorite location:
* save locations by posting to `/api/v1/favorites`
* delete locations by deleting to `/api/v1/favorites`
* view the current weather of all my favorite locations by getting `/api/v1/favorites`

### Requirements
- [x] (#15) Create favorites table
- [x] (#16) Add user favorites endpoint
- [x] (#17) List user favorites endpoint
- [x] (#18) Remove user favorites endpoint

### LINKED ISSUE #18: User Story - Delete Favorite Locations
As a user,
When I make a DELETE request to `/api/v1/favorites` with the following body:
```
Content-Type: application/json
Accept: application/json

body:

{
  "location": "Denver, CO",
  "api_key": "<My API KEY>"
}
```

I should get a status 204 message.

- [x] Successful response
- [x] Unauthorized response
- [x] Missing parameters response

### PR DESCRIPTION
**Pull request by CoopTang**
* Add delete route for user Favorites

closes #18 
closes #14

### CODE REVIEW

**Comment by CoopTang** [comment]
@Not-Zorro can you look over this please?

File: routes/api/v1/favorites.js [hunk 1]
```diff
@@ -27,16 +26,28 @@ router.post('/', (request, response) => {
...
       if (user) {
         validFavoriteResponse(request, response, user.id)
-        // let favorites = await database('favorites').where({ user_id: user.id }).select().first()
-        // // get the current weather for every favorite
-        // let favoriteWeathers = await createFavorites(favorites)
-        // // return array of these objects
-        // response.status(200).json(favoriteWeathers)
+      }
+      else {
+        invalidUserResponse(response);
```
Comment (davidlopez) [inline routes/api/v1/favorites.js:35]: I like the use of these helper methods

File: routes/api/v1/favorites.js [hunk 2]
```diff
@@ -27,16 +26,28 @@ router.post('/', (request, response) => {
...
+      }
+    })
+    .catch((error) => {
+      response.status(500).json({ error });
+    });
+  });
+
+router.delete('/', (request, response) => {
+  database('users').where({ apiKey: request.body.api_key }).select().first()
```
Comment (davidlopez) [inline routes/api/v1/favorites.js:44]: I believe you can remove the `.select()` from this line

**Review by davidlopez — COMMENTED** [review]
Great job overall, just remove that unneeded `.select()` then should be good to go.

**Pull request merged**